commons/src/main/java/org/infinispan/commons/marshall/WrappedByteArray.java

```diff
@@ -16,16 +16,9 @@
  */
 public class WrappedByteArray implements WrappedBytes {
    private final byte[] bytes;
-   private transient int hashCode;
```

Comment:
What was the motivation here?  Just to save 8 bytes per instance?  The hashCode will be called on every node at least once since it has to be actually put into the container.  Is this to avoid calling hashCode for values?

Sadly for off heap the hashCode is always done even for values before hand and caching it can save a lot of time, since it is required for values to be computed.